### PR TITLE
finished assignment7. thank you~

### PR DESCRIPTION
**Pull request by xkuang**

### CODE REVIEW

File: Assignment7.Rmd [hunk 1]
```diff
@@ -9,26 +9,58 @@ In the following assignment you will be looking at data from an one level of an
...
+  geom_histogram(data = D2[D2$variable == "post.test.score",], binwidth=0.01) +
+  geom_histogram(data = D2[D2$variable == "pre.test.score",], binwidth=0.01) + 
+  facet_wrap(~variable, scales = "free")
+#code reference: http://stackoverflow.com/questions/17271968/different-breaks-per-facet-in-ggplot2-histogram by user20650
 #Then visualize the relationships between variables
-
+D3 <- dplyr::select(D1, 2:6,8)
+COR <- cor(D3)
+corrplot(COR, order="AOE", method="circle", tl.pos="lt", type="upper",        
+         tl.col="black", tl.cex=0.6, tl.srt=45,
```
Comment (charles-lang) [inline Assignment7.Rmd]: You could also use the pairs() command to get a matrix of scatterplots

File: Assignment7.Rmd [hunk 2]
```diff
@@ -9,26 +9,58 @@ In the following assignment you will be looking at data from an one level of an
...
 #Try to capture an intution about the data and the relationships
-
+#The post test score is highly related to the number of messages(0.94), and pre test score/prior knowledge(67), assignment score(76). the post test score is not the simple cut-line for level-up. 
+#I guess level-up is based on a combination of the most related variables to level.up2: messages + post.test.score + av.assignment.score
 ```
 #Classification tree
 ```{r}
 #Create a classification tree that predicts whether a student "levels up" in the online course using three variables of your choice (As we did last time, set all controls to their minimums)
-
+c.tree1 <- rpart(level.up ~ messages + post.test.score + av.assignment.score, method="class", data=D1,control=rpart.control(minsplit = 1, minbucket = 1, cp = 0.001))
```
Comment (charles-lang) [inline Assignment7.Rmd]: Did you us your histogram matrix to make the decision about which variables to use?

File: Assignment7.Rmd (hunk 1 above)
Comment (xkuang) [inline Assignment7.Rmd]: thank you. pairs() helps~

File: Assignment7.Rmd (hunk 2 above)
Comment (xkuang) [inline Assignment7.Rmd]: I used the correlation plot to pick the most related variables to level.up. I think we can use the histogram to help make the decision as well if we could plot 2 variables' distribution on the same histogram panel to see the correlation pattern.

File: Assignment7.Rmd [hunk 3]
```diff
@@ -42,36 +74,112 @@ abline(0, 1, lty = 2)
...
+D1$threshold.pred1 <- ifelse(D1$pred <= threshold.pred1, "no","yes") #I had no idea about this step, searching online for hours, finally got hints from Edward and your collaboration. Thank you two.
+  #Now generate three diagnostics:
+table1 <- table(D1$level.up, D1$threshold.pred1)
+table1
+#     no yes
+#no  600   0
+#yes   0 400
+  accuracy1 <- (400+600)/(400+600) # 100%
+  precision1 <- 400/(400+0) #100%
+  recall1 <- 400/(400+0) #100%
```
Comment (xkuang) [inline Assignment7.Rmd]: I agree~

**Comment by xkuang** [comment]
Thanks. According to the classification tree Model 1, I think the program designers use average assignment and post-test score to make the decision on students' level-up status.

**Comment by charles-lang** [comment]
Sounds right to me, good work!

On Mon, Dec 05, 2016 at 10:42 AM Xiaoting Kuang

<
mailto:Xiaoting Kuang <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

Thanks. According to the classification tree Model 1, I think the program designers use average assignment and post-test score to make the decision on students' level-up status.

—

You are receiving this because you commented.

Reply to this email directly,
https://github.com/core-methods-in-edm/assignment7/pull/3#issuecomment-264887457
, or

**Pull request closed without merging**